### PR TITLE
Requesting a pull to mozilla:master from k88hudson:t2138

### PR DESCRIPTION
**Pull request by k88hudson**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/31f41fb128b889e/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @dseif received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/2154f137496130e/output.txt

File: src/editor/trackevent-editor.js [hunk 1]
```diff
@@ -2,10 +2,10 @@
  * If a copy of the MIT license was not distributed with this file, you can
  * obtain one at http://www.mozillapopcorn.org/butter-license.txt */
 
-define( [ "util/lang", "util/keys", "./base-editor",
+define( [ "util/lang", "util/keys", "util/time", "./base-editor",
```
Comment (dseif) [inline src/editor/trackevent-editor.js]: `( [` -> `([`

File: src/editor/trackevent-editor.js [hunk 2]
```diff
@@ -86,11 +86,15 @@ define( [ "util/lang", "util/keys", "./base-editor",
             appendTo: rootElement.querySelector( ".scrollbar-append-to" )
           });
         }
+
```
Comment (dseif) [inline src/editor/trackevent-editor.js]: can remove this line

File: src/editor/trackevent-editor.js [hunk 3]
```diff
@@ -153,6 +157,17 @@ define( [ "util/lang", "util/keys", "./base-editor",
...
     };
 
+    extendObject.showPluginPreview = function( trackEvent ) {
+      var startTime = trackEvent.popcornOptions.start,
+          endTime = trackEvent.popcornOptions.end,
+          currentTime = butter.currentTime,
+          accuracy = TimeUtils.timeAccuracy;
+      if ( currentTime < startTime || currentTime > endTime ) {
+        // Account for accuracy
+        butter.currentTime = Math.ceil( startTime * Math.pow( 10, accuracy - 1 ) ) / Math.pow( 10, accuracy - 1 ) ;
```
Comment (dseif) [inline src/editor/trackevent-editor.js]: Mind storing `Math.pow( 10, accuracy - 1 )` in a variable so we don't have to do it twice here

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/31f41fb128b889e/output.txt

Total script time: 1.74 mins

Server is running on http://pr888.butyr.org/templates/basic/
Run the Test Suite: http://pr888.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/2154f137496130e/output.txt

Total script time: 2.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brett-etherworks received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/23bd434164f77c3/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/23bd434164f77c3/output.txt

Total script time: 0.66 mins

Server is running on http://pr888.butyr.org/templates/basic/
Run the Test Suite: http://pr888.butyr.org/test/

**Pull request merged**